src/components/HeaderItem/index.jsx

```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import { Typography } from "@mui/material";
+
+export default function HeaderItem({
+  ButtonComponent,
+  handleItemClick,
+  handleCloseNavigation,
+  selectedButton,
+  route,
+  page,
+}) {
+  return (
+    <ButtonComponent
+      onClick={() => {
+        handleItemClick(route, page);
+        if (handleCloseNavigation !== undefined) {
```

Comment:
you can write check for underfined like that: 
`if(handleCloseNavigation) {
}`

if clause will work only if condition inside is not falsy value (null, undefined, false, 0, NaN, etc)


your condition is totally fine too, you can leave it if you want, it will still work how it's supposed to 👍
but usually you write such condition when you need to check specifically for undefined and nothing else :)